java-checks/src/main/java/org/sonar/java/checks/NoSonarCheck.java

```diff
@@ -36,18 +37,31 @@
 public class NoSonarCheck extends IssuableSubscriptionVisitor {
 
   private static final String PATTERN = "NOSONAR";
+  private static final String PATTERN_ONLY_WHEN_NO_DETAILS = "^[/\\*\\s]*NOSONAR[/\\*\\s]*$";
   private static final String MESSAGE = "Is //NOSONAR used to exclude false-positive or to hide real quality flaw ?";
 
   private final CommentContainsPatternChecker checker = new CommentContainsPatternChecker(this, PATTERN, MESSAGE);
 
+  @RuleProperty(
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Reorder annotation arguments to match the order of declaration. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3340)